### PR TITLE
Move Proposal/SubmittedProposal fields from SQL to plone objects.

### LINKED ISSUE #1357: Versionierung von Anträgen.
Ein Änderungsverlauf von Anträgen wäre ein Nice-To-Have.

Primär müsste ein Vergleich möglich sein zwischen eingereichtem Antrag und protokolliertem/behandeltem Antrag.

### LINKED ISSUE #3025: Reduce Proposal SQL dependencies in Dossier
We want to move primary data back from `SQL` to `Plone`. As a first step we eliminate SQL dependencies on the dossier side by only storing Proposal in SQL once it has been submitted.

Move Proposal from SQL back to Plone _(or die trying)_.  

- [x] move proposal text attributes to plone object
- [x] move submitted proposal text attributes to plone object
- [x] copy attributes on proposal submission
- [x] get rid of confusing submitted proposal builder, which does not build plone objects
- [x] improve builders, add mixin to transparently load model for `ModelContainer`
- [x] prepare support for builders handling sql/plone transparently
- [x] fix creating agenda items
- [x] fix protocol

### PR DESCRIPTION
**Pull request by deiferni**
Closes #3025

for the duration of the migration from sql to plone we will purposefully break multi-admin-unit support.

- copy proposal text attributes to `Proposal` plone objects
- also copy attributes to `SubmittedProposal`  plone objects
- drop copied columns in SQL
- copy object content on proposal submission, partially re-use `Transporter`
- get rid of confusing submitted proposal builder, which does not build plone objects and replace with a clean builder for `SubmittedProposal`
- improve builders, add mixin to transparently load model for `ModelContainer` and prepare support for builders handling sql/plone transparently

Also closes #1357.

### CODE REVIEW

**Comment by deiferni** [comment]
Yes, if we had a multi-admin-unit site setup id would be near impossible to upgrade all at once.  Luckily we know that we don't have that now nor in the near future 😉 , so we can take a shortcut there.

File: opengever/meeting/tests/test_unit_agenda_item.py [hunk 1]
```diff
@@ -1,15 +1,68 @@
...
+                               .having(title=u'Pr\xf6posal',
+                                       committee=self.committee.load_model()))
+        self.submitted_proposal = create(
+            Builder('submitted_proposal').submitting(self.proposal))
+        self.agenda_item = create(
+            Builder('agenda_item')
+            .having(proposal=self.proposal.load_model(),
+                    meeting=self.meeting))
+
+    def test_test_title_is_proposal_title_when_proposal_is_available(self):
```
Comment (phgross) [inline opengever/meeting/tests/test_unit_agenda_item.py]: test_test

File: opengever/core/upgrades/20170619120631_move_proposal_fields_to_plone_object/upgrade.py [hunk 2]
```diff
@@ -45,8 +51,18 @@ def migrate(self):
...
+        statement = admin_units_table.select().where(
+            admin_units_table.c.enabled==True)
+        return len(self.execute(statement).fetchall()) > 1
+
     def migrate_data(self):
-        for proposal in self.execute(proposal_table.select()):
+        proposals = self.execute(proposal_table.select()).fetchall()
+        if proposals:
+            msg = 'data migration supports only one admin-unit!'
+            assert not self.has_multiple_admin_units(), msg
```
Comment (phgross) [inline opengever/core/upgrades/20170619120631_move_proposal_fields_to_plone_object/upgrade.py]: This Upgradestep will raises when installing it on dev.onegovgever.ch. An assertion here is not an option.

File: opengever/meeting/tabs/proposallisting.py [hunk 3]
```diff
@@ -61,4 +61,5 @@ class ProposalTableSource(SqlTableSource):
     grok.implements(ITableSource)
     grok.adapts(ProposalListingTab, Interface)
 
-    searchable_columns = [Proposal.title, ]
+    # XXX make title searchable
```
Comment (phgross) [inline opengever/meeting/tabs/proposallisting.py]: That's done in an later commit, remove this comment.

File: opengever/meeting/tabs/submittedproposallisting.py [hunk 4]
```diff
@@ -42,4 +42,5 @@ class SubmittedProposalTableSource(SqlTableSource):
     grok.implements(ITableSource)
     grok.adapts(SubmittedProposalListingTab, Interface)
 
-    searchable_columns = [Proposal.title, ]
+    # XXX make title searchable
```
Comment (phgross) [inline opengever/meeting/tabs/submittedproposallisting.py]: That's done in an later commit, remove this comment.

File: opengever/meeting/model/proposal.py [hunk 5]
```diff
@@ -194,8 +184,8 @@ def css_class(self):
         return 'contenttype-opengever-meeting-proposal'
 
     def get_searchable_text(self):
-        searchable = filter(None, [self.title, self.initial_position])
-        return ' '.join([term.encode('utf-8') for term in searchable])
+        # XXX: we no longer have additional searchable text
```
Comment (deiferni) [inline opengever/meeting/model/proposal.py]: i found out that we can go even further and remove the whole searchable text extender for proposals, amended.

**Comment by deiferni** [comment]
@phgross updated, also i have changed the assertion in the upgrade step to avoid problems on `dev` and `test` as you mentioned

File: docs/HISTORY.txt [hunk 6]
```diff
@@ -23,6 +23,9 @@ Changelog
 - Add reference column to document listings. [tarnap]
 - Do not display templates without an assigned file in the CreateDocumentFromTemplate form. [tarnap]
 - Add dossier and subdossier link in task listings. [elioschmutz]
+- Include missing permission definitions from Products.CMFCore. [phgross]
```
Comment (jone) [inline docs/HISTORY.txt]: I think this does not belong here.

**Comment by deiferni** [comment]
@jone @phgross this one should be ready now, but requires your final approval 😉

**Pull request merged**